contracts/contracts/core/ProviderRegistry.sol

```diff
@@ -228,6 +234,35 @@ contract ProviderRegistry is
         emit BidderWithdrawSlashedAmount(msg.sender, amount);
     }
 
+    /**
+     * @dev Adds a verified BLS key to the provider's account.
+     * @param blsPublicKey The BLS public key to be added.
+     * @param signature The signature (96 bytes) used for verification.
+     */
+    function addVerifiedBLSKey(
+        bytes calldata blsPublicKey,
+        bytes calldata signature
+    ) external {
+        address provider = msg.sender;
+
+        require(providerRegistered[provider], "Provider not registered");
+        require(blsPublicKey.length == 48, "Public key must be 48 bytes");
+        require(signature.length == 96, "Signature must be 96 bytes");
+
+
+        bytes32 message = keccak256(abi.encodePacked(provider));
```

Comment:
For such authentication schemes, it is always a good idea to include some context into the signed message to prevent replay attacks and to prevent misuse of the signature in another context.

Here, only the provider address is signed. If some other place also simply expects a signature on this address, an attacker can use the signature obtained here to cause harm in the other place. Better would be to bind the signed message to this context, e.g., by signing Hash("BLS Key addition to mev-commit provider registry" + "mainnet" / "testnet" + mev-commit version + provider address)

Once the context is established, one still should bind the signature to the particular instance of the protocol execution to prevent replay attacks. E.g., if a provider wants to move this BLS key to another provider account since somebody got access to the old one, the holder of the original account can register again with the same signature used previously. This can be prevented by some challenge-response scheme, or a time-stamp, e.g., in the form of the mev-commit chain height.

Reply:
That's a great suggestion! Thanks~

Reply:
I think for now this should be fine, we can update this in a later iteration. The provider address generated for the user in mev-commit is singularly used as an EOA account in mev-commit. We're also simply authenticating the BLS public key to be tied to the EOA account.  So I don't see this being an issue, at least for now. If we increase the use-cases for BLS signatures, we can move onto this approach.